lib/widgets/gallery_view.dart

```diff
@@ -0,0 +1,258 @@
+import 'package:flutter/material.dart';
+import 'package:workoutpersonalizer_frontend/constants/styles.dart';
+
+class Gallery extends StatefulWidget {
+  final List<String> exerciseList;
+  final List<String> descriptionList;
+  final List<String> tagList;
+  final bool playlistGallery;
+
+  const Gallery(
+      {Key? key,
+      required this.exerciseList,
+      required this.playlistGallery,
+      required this.descriptionList,
+      required this.tagList})
+      : super(key: key);
+
+  @override
+  GalleryState createState() => GalleryState();
+}
+
+class GalleryState extends State<Gallery> {
+  List<String> exercisesToDisplay = [
+    'Pushup',
+    'Crunch',
+    'Neck Circles',
+    'Hamstring Stretch',
+    'Plank',
+    'Hip Flexor Stretch',
+  ];
+
+  void addToWorkouts(value) {
+    print("this is value " + value);
+  }
+
+  void addToLibrary() {
+    print("add to library");
+  }
+
+  List<String> getDisplayItems(text, exerciseList) {
+    List<String> ret = [];
+    exerciseList.forEach((String exercise) {
+      if (exercise.toLowerCase().contains(text)) {
+        ret.add(exercise);
+      }
+    });
+    return ret;
+  }
+
+  Widget buildGrid(BuildContext context, exerciseList, descriptionList, tagList,
+      playlistGallery) {
+    //TOO: figure out userEffect but in flutter to get rid of duplicate list
+    // exercisesToDisplay = exerciseList;
+    return GridView.builder(
+        gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+            maxCrossAxisExtent: 300,
+            childAspectRatio: 8 / 7,
+            crossAxisSpacing: 20,
+            mainAxisSpacing: 20),
+        itemCount: exercisesToDisplay.length,
+        itemBuilder: (context, index) {
+          return GestureDetector(
```

Comment:
Could you split this into it's own separate `ExerciseCard` class? You can put it in `lib/widgets/explore/exercise_card.dart`. It would also be great if you could define an `Exercise` model in `lib/models/exercise.dart` and instantiate the exercise cards using the model. See my PR for reference: https://github.com/ubclaunchpad/workoutpersonalizer-frontend/pull/2/files

Reply:
@ltan02 used the gallery thing in his creator's corner page and has created the file as well as the Exercise model. I think we will place it in a more central location rather than in the /explore directory since we are sharing this widget.